src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/jobs/CourtHearingLinkReminderJobTest.kt

```diff
@@ -6,64 +6,108 @@ import org.mockito.kotlin.doReturn
+    runJobOn(THURSDAY)
     verify(bookingFacade).courtHearingLinkReminder(courtBooking(), getServiceAsUser())
   }
 
   @Test
-  fun `should not call court hearing link reminder for bookings with video URL`() {
-    val booking = courtBooking().withMainCourtPrisonAppointment()
+  fun `should call court hearing link reminder on Friday for valid court bookings`() {
+    whenever(prisonAppointmentRepository.findAllActivePrisonAppointmentsOnDate(FRIDAY.plusDays(3))) doReturn courtBookingWithoutUrl.appointments()
+    runJobOn(FRIDAY)
+    verify(bookingFacade).courtHearingLinkReminder(courtBooking(), getServiceAsUser())
+  }
 
-    whenever(prisonAppointmentRepository.findAllActivePrisonAppointmentsOnDate(tomorrow())) doReturn booking.appointments()
+  @Test
+  fun `should not call court hearing link reminder on Saturdays`() {
+    runJobOn(SATURDAY)
+    verifyNoInteractions(bookingFacade)
+  }
 
-    courtHearingLinkReminderJob.block()
+  @Test
+  fun `should call court hearing link reminder on Sunday for valid court bookings`() {
+    whenever(prisonAppointmentRepository.findAllActivePrisonAppointmentsOnDate(SUNDAY.plusDays(1))) doReturn courtBookingWithoutUrl.appointments()
+    runJobOn(SUNDAY)
+    verify(bookingFacade).courtHearingLinkReminder(courtBooking(), getServiceAsUser())
+  }
 
+  @Test
+  fun `should not call court hearing link reminder for bookings with video URL`() {
+    val booking = courtBooking().withMainCourtPrisonAppointment()
+    whenever(prisonAppointmentRepository.findAllActivePrisonAppointmentsOnDate(TUESDAY)) doReturn booking.appointments()
+    runJobOn(MONDAY)
     verify(bookingFacade, never()).courtHearingLinkReminder(any(), any())
   }
 
   @Test
   fun `should not call court hearing link reminder for non-court bookings`() {
     val booking = probationBooking().withProbationPrisonAppointment()
-
-    whenever(prisonAppointmentRepository.findAllActivePrisonAppointmentsOnDate(tomorrow())) doReturn booking.appointments()
-
-    courtHearingLinkReminderJob.block()
-
+    whenever(prisonAppointmentRepository.findAllActivePrisonAppointmentsOnDate(TUESDAY)) doReturn booking.appointments()
+    runJobOn(MONDAY)
     verify(bookingFacade, never()).courtHearingLinkReminder(any(), any())
   }
 
   @Test
   fun `should not call court hearing link reminder if court is disabled`() {
     val booking = courtBooking(court = court(enabled = false)).withMainCourtPrisonAppointment()
-
-    whenever(prisonAppointmentRepository.findAllActivePrisonAppointmentsOnDate(tomorrow())) doReturn booking.appointments()
-
-    courtHearingLinkReminderJob.block()
-
+    whenever(prisonAppointmentRepository.findAllActivePrisonAppointmentsOnDate(TUESDAY)) doReturn booking.appointments()
+    runJobOn(MONDAY)
     verify(bookingFacade, never()).courtHearingLinkReminder(any(), any())
   }
+
+  private fun runJobOn(date: LocalDate) {
+    CourtHearingLinkReminderJob(prisonAppointmentRepository, bookingFacade) { date.atStartOfDay() }.runJob()
```

Comment:
This is setting the date on the TimeSource passed into the job making it possible to test each day.